platform/socket.cpp

```diff
@@ -0,0 +1,91 @@
+#include "socket.hpp"
+
+#include "base/assert.hpp"
+
+#include "std/algorithm.hpp"
+
+namespace platform {
+
+TestSocket::TestSocket()
+  : m_isConnected(false)
+{
+}
+
+TestSocket::~TestSocket()
+{
+  m_isConnected = false;
+}
+
+bool TestSocket::Open(string const & host, uint16_t port)
+{
+  m_isConnected = true;
+  return true;
+}
+
+void TestSocket::Close()
+{
+  m_isConnected = false;
+}
+
+bool TestSocket::Read(uint8_t * data, uint32_t count)
+{
+  if (!m_isConnected)
+    return false;
+
+  lock_guard<mutex> lg(m_inputMutex);
```

Comment:
They are exists.
I will use this implementation in unit tests.

Client is gps location reporter.
Server is emulation calls in test.
Gps reporter doesn't know the socket is not real. Reporter have own working thread, it makes socket calls from his own thread.
Server emulation calls from main test thread to check contents of the socket.